pkg/kubelet/dockershim/network/cni/cni_others.go

```diff
@@ -61,19 +63,53 @@ func (plugin *cniNetworkPlugin) platformInit() error {
 // TODO: Use the addToNetwork function to obtain the IP of the Pod. That will assume idempotent ADD call to the plugin.
 // Also fix the runtime's call to Status function to be done only in the case that the IP is lost, no need to do periodic calls
 func (plugin *cniNetworkPlugin) GetPodNetworkStatus(namespace string, name string, id kubecontainer.ContainerID) (*network.PodNetworkStatus, error) {
+
+	podIPs := plugin.GetPodIPs(id)
+	if podIPs != nil && len(podIPs) > 0 {
+		klog.V(3).Infof("get pod ip %v from plugin", podIPs)
+		return &network.PodNetworkStatus{IP: podIPs[0], IPs: podIPs}, nil
+	}
+	cninetwork := plugin.getDefaultNetwork()
+	cniNet := cninetwork.CNIConfig
+	netConfList := cninetwork.NetworkConfig
 	netnsPath, err := plugin.host.GetNetNS(id.ID)
 	if err != nil {
 		return nil, fmt.Errorf("CNI failed to retrieve network namespace path: %v", err)
 	}
 	if netnsPath == "" {
 		return nil, fmt.Errorf("Cannot find the network namespace, skipping pod network status for container %q", id)
 	}
+	rt, err := plugin.buildCNIRuntimeConf(name, namespace, id, netnsPath, nil, nil)
+	if err != nil {
+		klog.Errorf("Error get pod network status when building cni runtime conf: %v", err)
+		return nil, err
+	}
+	res, err := cniNet.GetNetworkListCachedResult(netConfList, rt)
 
-	ips, err := network.GetPodIPs(plugin.execer, plugin.nsenterPath, netnsPath, network.DefaultInterfaceName)
+	pdesc := PodDesc(namespace, name, id)
+	if res == nil {
+		klog.V(3).Info("Cached result doesn't exists for %s", pdesc)
+		return nil, nil
+	}
 	if err != nil {
+		klog.Errorf("Error get cached result %s for network %s/%s: %v", pdesc,
+			netConfList.Plugins[len(netConfList.Plugins) - 1].Network.Type,
+			netConfList.Plugins[len(netConfList.Plugins) - 1].Network.Name, err)
```

Comment:
@AlexeyPerevalov the result is for the whole network list, not just a single plugin, so it doesn't make sense to log a plugin type here. Also, you can simply do "netConfList.Name" becuase CNI ensures that the network Name in the NetworkConfigList object is the same for all plugins in the list.